Instalment adjust to work with decimal floats

dariodinizg opened the pull request:
##### What this PR made?
UPDATED:
Model and function treatment to work with decimals within a delta 0.08 cents

OLD:
The adjustment was an instalment modification, but reading the design, it should be an increase or decrease in loan rate. 
This fix provide different values for instalment.

##### Where could the review begin?
Within calculator/models.py 
and the test file for models.

##### How this could be tested manually?
$ python manage.py test -v2 

##### Ref
#32

dariodinizg committed:
Important Fix: Adjustment in loan.rate

julianeventeu commented on calculator/models.py:
```diff
@@ -73,29 +73,28 @@ class Loan(models.Model):
     )
 
-    def _instalment_adjustment(self):
+    def _rate_adjustment(self):
```
I think the new value rate should be saved on the database, maybe using the method save like installment.
Something like this:

def save(self, *args, **kwargs):
        self.rate = self._rate_adjustment()
        self.instalment += self.calculate_instalment()
        super(Loan, self).save(*args, **kwargs)

And so you can remove the call method _rate_adjustment from calculate_instalment.

dariodinizg replied:
I tried to keep both independent, but since we need the adjust value to find instalment value, its not possible to separate them. 

And I implemented your suggestion about tracking the auto adjust in the database, so I created a field for it. I think is better to save this in another column than just alter the rate for the loan, cause altering automatic will give the user a false impression about his input. User could type 5% and receive a 4.98% instead.

julianeventeu commented:
The point about check decimals on installment x value total loan you didn't make in this PR.
Are you doing this in other PR? Because you defined this feature on issue #32 

_"Due to float places, its possible that a loan's value be different than the sum of it's instalments. We need to make sure it wont happen."_

dariodinizg committed:
Update models.py

Add: 
- Field for rate adjust
- Format constrains for decimal
- Changed term field to decimal field.

dariodinizg committed:
Update test_models.py

Add:
- Tests for float places in instalment calculations.

dariodinizg committed:
rebasing

dariodinizg committed:
rebased

dariodinizg committed:
rebased

dariodinizg committed:
Clean Decimal import line

dariodinizg commented:
> The point about check decimals on installment x value total loan you didn't make in this PR.
> Are you doing this in other PR? Because you defined this feature on issue #32
> 
> _"Due to float places, its possible that a loan's value be different than the sum of it's instalments. We need to make sure it wont happen."_

At the moment I wrote this issue I was thinking about grouping areas of implementations, so I put it here since both issues was related to instalment. I should keep then separated though...
Thanks for the remainder..

dariodinizg committed:
Cleaning code

dariodinizg committed:
Done

cesarbruschetta committed:
Merge remote-tracking branch 'origin/development' into tk_32

cesarbruschetta committed:
Merge branch 'tk_32' of github.com:cesarbruschetta/api-fintech into tk_32
